src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx

```diff
@@ -0,0 +1,296 @@
+  const [newOrgName, setNewOrgName] = useState('')
+  const providerNames = Object.keys(providerMap)
+  const numProviders = providerNames.length
+  const providerHelpText =
+    numProviders > 1
+      ? ProviderSelectMessage.MultiProvider
+      : ProviderSelectMessage.SingleProvider
+
+  // Error Handling State
+  const [createOrgButtonStatus, setCreateOrgButtonStatus] = useState(
+    ComponentStatus.Disabled
+  )
+  const [networkErrorMsg, setNetworkErrorMsg] = useState(
+    OrgOverlayNetworkError.None
+  )
+  const [validationMsg, setValidationMsg] = useState(
+    OrgOverlayValidationError.None
+  )
+
+  // Spinner State
+  const loadingComplete = Boolean(currentProvider) && Boolean(currentRegion)
+  const loadingState = loadingComplete
+    ? RemoteDataState.Done
+    : RemoteDataState.Loading
+
+  // Event Handlers
+  const handleCreateOrg = async () => {
+    try {
+      setCreateOrgButtonStatus(ComponentStatus.Loading)
+      await createNewOrg({
+        orgName: newOrgName,
+        provider: currentProvider,
+        region: currentRegion,
+      })
+      setCreateOrgButtonStatus(ComponentStatus.Default)
+      onClose()
+      dispatch(notify(orgCreateSuccess()))
+      // Will make another API call to the allowance endpoint in PR resolving this issue.
+      // https://github.com/influxdata/ui/issues/6267
+      // Need a parallel update to global header state to ensure list of orgs remains updated.
+      dispatch(getQuartzOrganizationsThunk(currentAccountId))
+    } catch (err) {
+      if (err instanceof UnprocessableEntityError) {
+        setNetworkErrorMsg(OrgOverlayNetworkError.NameConflictError)
+      } else if (err instanceof UnauthorizedError) {
+        setNetworkErrorMsg(OrgOverlayNetworkError.UnauthorizedError)
+      } else if (err instanceof ForbiddenError) {
+        setNetworkErrorMsg(OrgOverlayNetworkError.OrgLimitError)
+      } else {
+        setNetworkErrorMsg(OrgOverlayNetworkError.DefaultError)
+      }
+      setCreateOrgButtonStatus(ComponentStatus.Error)
+    }
+  }
+
+  const handleSelectProvider = (
+    providerID: OrganizationCreateRequest['provider']
+  ) => {
+    setCurrentProvider(providerID)
+    const defaultRegionId = providerMap[providerID][0].regionId
+    setCurrentRegion(defaultRegionId)
+  }
+
+  const handleValidateOrgName = (proposedOrgName: string) => {
+    setNetworkErrorMsg(OrgOverlayNetworkError.None)
+    setNewOrgName(proposedOrgName)
+
+    if (proposedOrgName.trim() === '') {
+      setCreateOrgButtonStatus(ComponentStatus.Disabled)
+      return
+    }
+
+    if (organizations.find(oldOrg => oldOrg.name === proposedOrgName)) {
+      setValidationMsg(OrgOverlayValidationError.NameConflictError)
+      setCreateOrgButtonStatus(ComponentStatus.Disabled)
+    } else {
+      setValidationMsg(OrgOverlayValidationError.None)
+      setCreateOrgButtonStatus(ComponentStatus.Default)
+    }
+  }
+
+  // Ajax requests
+  useEffect(() => {
```

Comment:
this is more of a personal preference than anything but I'd move these useEffect hooks to the top of the component where the states are being set